File: Lua/1021.lua
n = tonumber(io.read())

n = math.floor(n * 100)

x_100 = math.floor(n / (100*100))
r_100 = n % (100*100)

x_50 = math.floor(r_100 / (50*100))
r_50 = r_100 % (50*100)

x_20 = math.floor(r_50 / (20*100))
r_20 = r_50 % (20*100)

x_10 = math.floor(r_20 / (10*100))
r_10 = r_20 % (10*100)

x_5 = math.floor(r_10 / (5*100))
r_5 = r_10 % (5*100)

x_2 = math.floor(r_5 / (2*100))
r_2 = r_5 % (2*100)

x_1 = math.floor(r_2 / (1*100))
r_1 = r_2 % (1*100)

x_050 = math.floor(r_1 / (0.5*100))
r_050 = r_1 % (0.5*100)

x_025 = math.floor(r_050 / (0.25*100))
r_025 = r_050 % (0.25*100)

x_010 = math.floor(r_025 / (0.1*100))
r_010 = r_025 % (0.1*100)

x_005 = math.floor(r_010 / (0.05*100))
r_005 = r_010 % (0.05*100)

x_001 = math.floor((n % (0.05*100)) / (0.01*100))

print('NOTAS:')
print(x_100 .. ' nota(s) de R$ 100.00')
print(x_50 .. ' nota(s) de R$ 50.00')
print(x_20 .. ' nota(s) de R$ 20.00')
print(x_10 .. ' nota(s) de R$ 10.00')
print(x_5 .. ' nota(s) de R$ 5.00')
print(x_2 .. ' nota(s) de R$ 2.00')
print('MOEDAS:')
print(x_1 .. ' moeda(s) de R$ 1.00')
print(x_050 .. ' moeda(s) de R$ 0.50')
print(x_025 .. ' moeda(s) de R$ 0.25')
print(x_010 .. ' moeda(s) de R$ 0.10')
print(x_005 .. ' moeda(s) de R$ 0.05')
print(x_001 .. ' moeda(s) de R$ 0.01')

File: C/1021.c
#include <stdio.h>

int main() { 
    double x;
    int n, x_100, r_100, x_50, r_50, x_20, r_20, x_10, r_10, x_5, r_5, x_2, r_2, x_1, r_1, x_050, r_050, x_025, r_025, x_010, r_010, x_005, r_005, x_001, r_001;
    
    scanf("%lf", &x);

    n = (int) (x * 100);

    x_100 = n/(100*100);
    r_100 = n%(100*100);

    x_50 = r_100/(50*100);
    r_50 = r_100%(50*100);

    x_20 = r_50/(20*100);
    r_20 = r_50%(20*100);

    x_10 = r_20/(10*100);
    r_10 = r_20%(10*100);

    x_5 = r_10/(5*100);
    r_5 = r_10%(5*100);

    x_2 = r_5/(2*100);
    r_2 = r_5%(2*100);

    x_1 = r_2/(1*100);
    r_1 = r_2%(1*100);

    x_050 = r_1/(50);
    r_050 = r_1%(50);

    x_025 = r_050/(25);
    r_025 = r_050%(25);

    x_010 = r_025/(10);
    r_010 = r_025%(10);

    x_005 = r_010/(5);
    r_005 = r_010%(5);

    x_001 = r_005;

    printf("NOTAS:\n");
    printf("%d nota(s) de R$ 100.00\n", x_100);
    printf("%d nota(s) de R$ 50.00\n", x_50);
    printf("%d nota(s) de R$ 20.00\n", x_20);
    printf("%d nota(s) de R$ 10.00\n", x_10);
    printf("%d nota(s) de R$ 5.00\n", x_5);
    printf("%d nota(s) de R$ 2.00\n", x_2);
    printf("MOEDAS:\n");
    printf("%d moeda(s) de R$ 1.00\n", x_1);
    printf("%d moeda(s) de R$ 0.50\n", x_050);
    printf("%d moeda(s) de R$ 0.25\n", x_025);
    printf("%d moeda(s) de R$ 0.10\n", x_010);
    printf("%d moeda(s) de R$ 0.05\n", x_005);
    printf("%d moeda(s) de R$ 0.01\n", x_001);

    return 0;
}